test/e2e_node/cpu_manager_test.go

```diff
@@ -557,6 +559,153 @@ func runMultipleCPUContainersGuPod(ctx context.Context, f *framework.Framework)
+	pod1 = e2epod.NewPodClient(f).CreateSync(ctx, pod1)
+	cleanupPods = append(cleanupPods, pod1)
+
+	ginkgo.By("checking if the expected cfs quota was assigned (GU pod, exclusive CPUs, unlimited)")
+
+	expectedQuota := "100000"
+	if disabledCPUQuotaWithExclusiveCPUs {
+		expectedQuota = "max"
+	}
+	expCFSQuotaRegex := fmt.Sprintf("^%s %s\n$", expectedQuota, defaultPeriod)
+	err = e2epod.NewPodClient(f).MatchContainerOutput(ctx, pod1.Name, pod1.Spec.Containers[0].Name, expCFSQuotaRegex)
+	framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+		pod1.Spec.Containers[0].Name, pod1.Name)
+
+	ctnAttrs = []ctnAttribute{
+		{
+			ctnName:    "gu-container-cfsquota-enabled",
+			cpuRequest: "500m",
+			cpuLimit:   "500m",
+		},
+	}
+	pod2 = makeCPUManagerPod("gu-pod2", ctnAttrs)
+	pod2.Spec.Containers[0].Command = cfsCheckCommand
+	pod2 = e2epod.NewPodClient(f).CreateSync(ctx, pod2)
+	cleanupPods = append(cleanupPods, pod2)
+
+	ginkgo.By("checking if the expected cfs quota was assigned (GU pod, limited)")
+
+	expectedQuota = "50000"
+	expCFSQuotaRegex = fmt.Sprintf("^%s %s\n$", expectedQuota, defaultPeriod)
+	err = e2epod.NewPodClient(f).MatchContainerOutput(ctx, pod2.Name, pod2.Spec.Containers[0].Name, expCFSQuotaRegex)
+	framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+		pod2.Spec.Containers[0].Name, pod2.Name)
+
+	ctnAttrs = []ctnAttribute{
+		{
+			ctnName:    "non-gu-container",
+			cpuRequest: "100m",
+			cpuLimit:   "500m",
+		},
+	}
+	pod3 = makeCPUManagerPod("non-gu-pod3", ctnAttrs)
+	pod3.Spec.Containers[0].Command = cfsCheckCommand
+	pod3 = e2epod.NewPodClient(f).CreateSync(ctx, pod3)
+	cleanupPods = append(cleanupPods, pod3)
+
+	ginkgo.By("checking if the expected cfs quota was assigned (BU pod, limited)")
+
+	expectedQuota = "50000"
+	expCFSQuotaRegex = fmt.Sprintf("^%s %s\n$", expectedQuota, defaultPeriod)
+	err = e2epod.NewPodClient(f).MatchContainerOutput(ctx, pod3.Name, pod3.Spec.Containers[0].Name, expCFSQuotaRegex)
+	framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
+		pod3.Spec.Containers[0].Name, pod3.Name)
+
+	ctnAttrs = []ctnAttribute{
+		{
+			ctnName:    "gu-container-int-values",
+			cpuRequest: "1",
+			cpuLimit:   "1",
+		},
+		{
+			ctnName:    "gu-container-non-int-values",
+			cpuRequest: "500m",
+			cpuLimit:   "500m",
+		},
+	}
+	pod4 := makeCPUManagerPod("gu-pod4", ctnAttrs)
+	pod4.Spec.Containers[0].Command = cfsCheckCommand
+	pod4.Spec.Containers[1].Command = cfsCheckCommand
+	pod4 = e2epod.NewPodClient(f).CreateSync(ctx, pod4)
+	cleanupPods = append(cleanupPods, pod4)
+
+	ginkgo.By("checking if the expected cfs quota was assigned (GU pod, container 0 exclusive CPUs unlimited, container 1 limited)")
+
+	expectedQuota = "100000"
+	if disabledCPUQuotaWithExclusiveCPUs {
+		expectedQuota = "max"
+	}
+	expCFSQuotaRegex = fmt.Sprintf("^%s %s\n$", expectedQuota, defaultPeriod)
```

Comment:
You should also check the `gu-container-non-int-values` container here to ensure its correct

Reply:
resolved: see comment https://github.com/kubernetes/kubernetes/issues/70585#issuecomment-2518990304 had to split to check both pod-level cfs and container-level cfs